validations/src/validations.rs

```diff
@@ -1092,6 +1187,80 @@ pub fn validate_pkh_signature(
             }
             .into());
         }
+    } else {
+        log::error!("Normal Input can not spent a ScriptOutput");
```

Comment:
This branch will also be entered when the output is `None`, which can happen in some cases when trying to do a double spend. So convert the `if let` into a `match`, where the `None` case is just an empty `{}`.